core/src/main/scala/org/http4s/headers/Expires.scala

```diff
@@ -18,8 +13,14 @@ object Expires extends HeaderKey.Internal[Expires] with HeaderKey.Singleton {
   *
   * @param expirationDate the date of expiration
   */
-final case class Expires(expirationDate: HttpDate) extends Header.Parsed {
-  val key = `Expires`
-  override val value = Renderer.renderString(expirationDate)
-  override def renderValue(writer: Writer): writer.type = writer.append(value)
-}
+final case class Expires(expirationDate: java.time.Instant)
+object Expires {
+  implicit val expiresHeader: Header[Expires] =
+      new Header[Expires] {
+        val name = CaseInsensitiveString("Expires")
```

Comment:
I'm not entirely sure that `CaseInsensitiveString` has been a successful experiment, and we're about to double down on it.  What do other people think?